### ISSUE TITLE
npm install on mac failed

### ISSUE THREAD

**Issue by pmaganti** [tugrul/node-mcrypt#33]
Getting error like this when i tried to install mcrypt.
```
$ npm install mcrypt -g

> mcrypt@0.1.2 install /usr/local/lib/node_modules/mcrypt
> node-gyp rebuild --release

  CC(target) Release/obj.target/libmcrypt/lib/libmcrypt/modules/algorithms/3-way.o
In file included from ../lib/libmcrypt/modules/algorithms/3-way.c:14:
../lib/libmcrypt/lib/libdefs.h:55:11: fatal error: 'endian.h' file not found
# include <endian.h>
          ^
1 error generated.
make: *** [Release/obj.target/libmcrypt/lib/libmcrypt/modules/algorithms/3-way.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:807:12)
gyp ERR! System Darwin 14.4.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--release"
gyp ERR! cwd /usr/local/lib/node_modules/mcrypt
gyp ERR! node -v v0.10.28
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok 
npm ERR! Darwin 14.4.0
npm ERR! argv "node" "/usr/local/bin/npm" "install" "mcrypt" "-g"
npm ERR! node v0.10.28
npm ERR! npm  v2.10.1
npm ERR! code ELIFECYCLE

npm ERR! mcrypt@0.1.2 install: `node-gyp rebuild --release`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mcrypt@0.1.2 install script 'node-gyp rebuild --release'.
npm ERR! This is most likely a problem with the mcrypt package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild --release
npm ERR! You can get their info via:
npm ERR!     npm owner ls mcrypt
npm ERR! There is likely additional logging output above.

> mcrypt@0.1.2 preuninstall /usr/local/lib/node_modules/mcrypt
> rm -rf build/*
```

Tried solutions like:
1) installing nan
2) downgrading node-gyp to 1.0.1

**Comment by mikeljames** [comment]
+1 same issue for me.

**Comment by tugrul** [comment]
Same issue #31 Please use mcrypt@0.0.15 version. I'm working on libmcrypt source to port cpp.

**Comment by mikeljames** [comment]
Thanks @tugrul  :)

**Comment by pmaganti** [comment]
New error now when i tried install 0.0.15 version. Am i missing any dependencies?
Please help me @tugrul .

```
$ sudo npm install mcrypt@0.0.15 -g
\
> mcrypt@0.0.15 install /usr/local/lib/node_modules/mcrypt
> node-gyp rebuild --release

  CXX(target) Release/obj.target/mcrypt/src/mcrypt.o
In file included from ../src/mcrypt.cc:3:
../src/mcrypt.h:10:10: error: 'mcrypt.h' file not found with <angled> include; use "quotes" instead
#include <mcrypt.h>
         ^~~~~~~~~~
         "mcrypt.h"
../src/mcrypt.h:27:33: error: C++ requires a type specifier for all declarations
        template <int (*modify)(MCRYPT, void *, int)>
                                ^~~~~~
../src/mcrypt.h:57:9: error: unknown type name 'MCRYPT'
        MCRYPT mcrypt_;
        ^
../src/mcrypt.cc:17:15: error: use of undeclared identifier 'mcrypt_module_open'
    mcrypt_ = mcrypt_module_open(*algo, *algoDir, *mode, *modeDir);
              ^
../src/mcrypt.cc:25:25: error: unknown type name 'MCRYPT'
template <int (*modify)(MCRYPT mcrypt, void* target, int length)>
                        ^
../src/mcrypt.cc:68:9: error: use of undeclared identifier 'mcrypt_free'
        mcrypt_free(sizes);
        ^
../src/mcrypt.cc:88:5: error: use of undeclared identifier 'mcrypt_free'
    mcrypt_free(sizes);
    ^
../src/mcrypt.cc:108:5: error: use of undeclared identifier 'MCRYPT_FAILED'
    MCRYPT_MODULE_ERROR_CHECK(mcrypt)
    ^
../src/mcrypt.h:13:28: note: expanded from macro 'MCRYPT_MODULE_ERROR_CHECK'
    if (mcrypt->mcrypt_ == MCRYPT_FAILED) { \
                           ^
../src/mcrypt.cc:110:5: error: cannot initialize object parameter of type 'node::ObjectWrap' with an expression of type 'MCrypt'
    mcrypt->Wrap(args.This());
    ^~~~~~
../src/mcrypt.cc:124:5: error: use of undeclared identifier 'MCRYPT_FAILED'
    MCRYPT_MODULE_ERROR_CHECK(mcrypt)
    ^
../src/mcrypt.h:13:28: note: expanded from macro 'MCRYPT_MODULE_ERROR_CHECK'
    if (mcrypt->mcrypt_ == MCRYPT_FAILED) { \
                           ^
../src/mcrypt.cc:212:5: error: use of undeclared identifier 'MCRYPT_FAILED'
    MCRYPT_MODULE_ERROR_CHECK(mcrypt)
    ^
../src/mcrypt.h:13:28: note: expanded from macro 'MCRYPT_MODULE_ERROR_CHECK'
    if (mcrypt->mcrypt_ == MCRYPT_FAILED) { \
                           ^
../src/mcrypt.cc:222:40: error: use of undeclared identifier 'mcrypt_generic'
        cipherText = mcrypt->transform<mcrypt_generic>(*value, &length, &result);
                                       ^
../src/mcrypt.cc:222:55: error: expected unqualified-id
        cipherText = mcrypt->transform<mcrypt_generic>(*value, &length, &result);
                                                      ^
../src/mcrypt.cc:227:40: error: use of undeclared identifier 'mcrypt_generic'
        cipherText = mcrypt->transform<mcrypt_generic>(node::Buffer::Data(args[0]), &length, &result); 
                                       ^
../src/mcrypt.cc:227:55: error: expected unqualified-id
        cipherText = mcrypt->transform<mcrypt_generic>(node::Buffer::Data(args[0]), &length, &result); 
                                                      ^
../src/mcrypt.cc:234:29: error: use of undeclared identifier 'mcrypt_strerror'
        const char* error = mcrypt_strerror(result);
                            ^
../src/mcrypt.cc:254:5: error: use of undeclared identifier 'MCRYPT_FAILED'
    MCRYPT_MODULE_ERROR_CHECK(mcrypt)
    ^
../src/mcrypt.h:13:28: note: expanded from macro 'MCRYPT_MODULE_ERROR_CHECK'
    if (mcrypt->mcrypt_ == MCRYPT_FAILED) { \
                           ^
../src/mcrypt.cc:264:40: error: use of undeclared identifier 'mdecrypt_generic'
        cipherText = mcrypt->transform<mdecrypt_generic>(*value, &length, &result);
                                       ^
../src/mcrypt.cc:264:57: error: expected unqualified-id
        cipherText = mcrypt->transform<mdecrypt_generic>(*value, &length, &result);
                                                        ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/mcrypt/src/mcrypt.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Darwin 14.5.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--release"
gyp ERR! cwd /usr/local/lib/node_modules/mcrypt
gyp ERR! node -v v0.12.7
gyp ERR! node-gyp -v v2.0.1
gyp ERR! not ok 
npm ERR! Darwin 14.5.0
npm ERR! argv "node" "/usr/local/bin/npm" "install" "mcrypt@0.0.15" "-g"
npm ERR! node v0.12.7
npm ERR! npm  v2.12.1
npm ERR! code ELIFECYCLE

npm ERR! mcrypt@0.0.15 install: `node-gyp rebuild --release`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mcrypt@0.0.15 install script 'node-gyp rebuild --release'.
npm ERR! This is most likely a problem with the mcrypt package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild --release
npm ERR! You can get their info via:
npm ERR!     npm owner ls mcrypt
npm ERR! There is likely additional logging output above.

> mcrypt@0.0.15 preuninstall /usr/local/lib/node_modules/mcrypt
> rm -rf build/*

```

**Comment by tugrul** [comment]
@pmaganti yeah you need to install libmcrypt and libmcrypt-dev

**Comment by pmaganti** [comment]
Am i dumb guy here :(

I did not find libmcrypt and libmcrypt-dev using npm.
When i tried using brew it installed mcrypt-2.6.8.

Sorry to bother you again @tugrul

**Comment by tugrul** [comment]
@pmaganti 

Debian / Ubuntu:
`sudo apt-get install libmcrypt libmcrypt-dev`

**Comment by gsklee** [comment]
+1

**Comment by rhlsthrm** [comment]
I am having the same issue here.  Trying to run on Mac OSX using brew.

**Comment by ShaynaCummings** [comment]
I was able to resolve this issue on OS X El Capitan (running node v0.10.40) with the following:
`xcode-select --install # I needed to update xcode tools and manually accept TOS` 
`node-gyp rebuild`
`npm install mcrypt`

**Comment by tugrul** [comment]
Thank you @ShaynaCummings. Is there an anybody who profit by this solution for other versions?

**Comment by lmj0011** [comment]
This was resolved for on Xubuntu 16.04 with the following command:
`sudo apt install node-gyp` :+1:
